core-api/src/services/checkoutSlot.service.ts

```diff
@@ -0,0 +1,57 @@
+import db from "../config/database";
+import { Place } from "../entities/place.entity";
+import { ServerError, ErrorCode } from "../utils/serverError";
+import { SafeCheckoutSlot, CheckoutSlot } from "../entities/checkoutSlot.entity";
+import messages from "../utils/messages";
+
+/**
+ * @namespace Services
+ * @class PlaceService
+ */
+export class CheckoutSlotService {
+
+    public static async findAll(): Promise<SafeCheckoutSlot[]> {
+        const repository = db.getRepository(CheckoutSlot);
+
+        try {
+            const slots: CheckoutSlot[] = await repository.find({ relations: ["place"] });
+            return slots.map(slot => slot.toSafe());
+        } catch (e) {
+            throw new ServerError(e.message, ErrorCode.DATABASE_ERROR);
+        }
+    }
+
+    public static async findByPlace(placeId: number): Promise<SafeCheckoutSlot[] | null> {
+        const repository = db.getRepository(CheckoutSlot);
+
+        try {
+            const slots: CheckoutSlot[] = await repository.createQueryBuilder("checkout_slot")
+                .leftJoinAndSelect("checkout_slot.place", "place")
+                .where("place.id = :placeId")
+                .setParameters({ 'placeId': placeId })
+                .getMany();
+
+            return slots.map(slot => slot.toSafe());
+        } catch (e) {
+            throw new ServerError(e.message, ErrorCode.DATABASE_ERROR);
+        }
+    }
+
+    public static async createCheckoutSlot(placeId: number, start: Date, end: Date): Promise<SafeCheckoutSlot> {
+        const placeRepository = db.getRepository(Place);
+        const slotRepository = db.getRepository(CheckoutSlot);
+
+        try {
+            const place: Place = await placeRepository.findOne(placeId);
```

Comment:
I believe it would be better you called the PlaceService, instead of calling the place repository directily. There's already a method there.

Reply:
I agree. This is due to the fact that service methods are returning "safe" objects, in this case I don't want them and accessing the repository was a simples workaround.

Reply:
You can create a method in PlaceService to findByIdWithoutSafety and call here.

Reply:
Got it. Actually there's a method called findByIdWithoutSafety in the place service. It would fit your need.

Reply:
done